.github/workflows/deploy_website.yml

```
@@ -0,0 +1,42 @@
+name: Deploy Website
+run-name: Deploy Website
+
+on:
+  push:
+    branches: [ main ]
+  workflow_dispatch: # Allows to run this workflow manually from the Actions tab.
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write # Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages.
+
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  build:
+      runs-on: ubuntu-latest
+      steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup pages
+        uses: actions/configure-pages@v3
+      - name: Build
+        uses: actions/jekyll-build-pages@v1
+        with:
+            source: ./docs
+      - name: Upload artifacts
+        uses: actions/upload-pages-artifact@v1
+  
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Deploy Website
+        id: deployment
+        uses: actions/deploy-pages@v2
```

nit: please include a new line at EOF.

```suggestion
        uses: actions/deploy-pages@v2

```

---

I added it manually.